nodebuilder/p2p/pubsub.go

```diff
@@ -109,12 +110,23 @@ func topicScoreParams(network Network) map[string]*pubsub.TopicScoreParams {
 	return mp
 }
 
-func peerScoreParams(isBootstrapper bool, bootstrappers Bootstrappers) *pubsub.PeerScoreParams {
+func peerScoreParams(isBootstrapper bool, bootstrappers Bootstrappers, cfg Config) *pubsub.PeerScoreParams {
 	bootstrapperSet := map[peer.ID]struct{}{}
 	for _, b := range bootstrappers {
 		bootstrapperSet[b.ID] = struct{}{}
 	}
 
+	var ipColocFactWl []*net.IPNet = nil
+	if len(cfg.IPColocationFactorWhitelist) > 0 {
+		for _, strIp := range cfg.IPColocationFactorWhitelist {
+			_, ipNet, err := net.ParseCIDR(strIp)
+			if err != nil {
+				panic(err)
+			}
+			ipColocFactWl = append(ipColocFactWl, ipNet)
+		}
+	}
```

Comment:
Some simplifications:
* No need to check for an empty slice before iterating, as the range is noop in such cases
* No nee to explicitly set nil when using var
```suggestion
	var ipColocFactWl []*net.IPNet
	for _, strIp := range cfg.IPColocationFactorWhitelist {
		_, ipNet, err := net.ParseCIDR(strIp)
		if err != nil {
			panic(err)
		}
		ipColocFactWl = append(ipColocFactWl, ipNet)
	}
```